lib/Sema/TypeCheckConstraints.cpp

```diff
@@ -451,6 +452,36 @@ resolveDeclRefExpr(UnresolvedDeclRefExpr *UDRE, DeclContext *DC) {
       diagnose(Loc, diag::use_unresolved_identifier, Name,
                UDRE->getName().isOperator())
         .highlight(Loc);
+
+      const char *buffer = Name.get();
+      std::string expectedIdentifier = std::string();
```

Comment:
Now that this is at the top level, you can use a SmallString instead of std::string here too.